### PR TITLE
Development

### PR DESCRIPTION
**Pull request by SebastianPetak**
@KasaiMagi Having a lot of trouble unit testing react applications and other new tools I'm using. I think I need to create more applications with the tools before I try unit testing them. This project will display something like what this [app](https://codepen.io/freeCodeCamp/full/eZGMjp) displays.

### CODE REVIEW

**Comment by tdbishop** [comment]
nice!

File: app/containers/CamperLeaderboard.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			If leaderboard is empty, return null so camperTable renders without data.
+			Otherwise fill the table with the appropriate data from leaderboard.
+		*/
+		let leaderboardRows;
+		if (_.isEmpty(this.state.leaderboard)) {
+			leaderboardRows = null;
+		} else {
+			leaderboardRows = this.state.leaderboard.map( (camperData, index) => {
+				return (
+					<tr key={index}>
```
Comment (tdbishop) [inline app/containers/CamperLeaderboard.js:58]: does `camperData`  have a more identifying field than the index here? Reason being, if you add/remove/reorder this list, the index will not change and not cause this rendering to update accordingly. The `tr` `key` should be unique to the piece of data here.

https://facebook.github.io/react/docs/lists-and-keys.html#keys

Comment (tdbishop) [reply app/containers/CamperLeaderboard.js:58]: looks like `camperData.username` might be perfect since you're concating that into a url on line 61 too.

Comment (SebastianPetak) [reply app/containers/CamperLeaderboard.js:58]: Oh, interesting. I see your point. So is the `key` the only thing React is tracking to know if it should update?

Comment (tdbishop) [reply app/containers/CamperLeaderboard.js:58]: I think it's the only thing but also get a feeling React is smarter than that... heh

**Pull request merged**